test/podFailover/workload/main.go

```diff
@@ -17,49 +17,175 @@ limitations under the License.
+
+	if podfailurecrd == nil {
+		podfailurecrd, err = createCrd(ctx, podFailoverClient, podName)
+		if err != nil {
+			klog.Errorf("Error occurred while getting or creating podfailure crd for pod %s: %v", podName, err)
+		}
+	} else {
+		downtime := calculateDowntime(podfailurecrd.Status.HeartBeat)
+		reportDowntime(ctx, clientset, downtime, podfailurecrd.Status.FailureType)
+	}
+
+	if *storageType != "" {
+		runStatefulWorkload(ctx, clientset)
+	} else {
+		runStatelessWorkload(ctx, clientset, podFailoverClient, podfailurecrd)
+	}
+}
+
+func runStatefulWorkload(ctx context.Context, clientset *kubernetes.Clientset) {
 	filePath := *mountPath + "/outfile"
 	klog.Infof("The file path is %s", filePath)
-	fi, err := os.Stat(filePath)
+	_, err := os.Stat(filePath)
 	if os.IsNotExist(err) {
 		_, err = os.Create(filePath)
 		if err != nil {
 			log.Fatalf("Error occurred while creating the file %s: %v", filePath, err)
 		}
-		fi, err = os.Stat(filePath)
+		_, err = os.Stat(filePath)
 		if err != nil {
 			klog.Errorf("Error occurred while getting the stats for file %s: %v", filePath, err)
 		}
 	}
-	fileModTime := fi.ModTime()
 
-	var timeDifference int64
-	if fi.Size() > 0 {
-		timeDifference = time.Now().Unix() - fileModTime.Unix()
-		klog.Infof("The downtime seen by the pod is %d", timeDifference)
+	file, err := os.OpenFile(filePath, os.O_RDWR, os.ModeAppend)
+	if err != nil {
+		log.Fatalf("Couldn't open the file %s", filePath)
+	}
+	// Write timestamp to the file every second
+	go logTimestamp(file)
+	defer file.Close()
 
-		client := &http.Client{}
+	preStopMux := http.NewServeMux()
+	preStopMux.HandleFunc("/cleanup", func(w http.ResponseWriter, r *http.Request) {
+		file.Close()
+	})
+	srv := &http.Server{Addr: ":9091", Handler: preStopMux}
+
+	func() { log.Fatal(srv.ListenAndServe()) }()
+
+}
+
+func runStatelessWorkload(ctx context.Context, clientset *kubernetes.Clientset, podFailoverClient *podfailure.PodFailoverClient, failureCrd *podfailure.PodFailure) {
+	go reportTimestamp(ctx, podFailoverClient, failureCrd)
+	preStopMux := http.NewServeMux()
+	preStopMux.HandleFunc("/cleanup", func(w http.ResponseWriter, r *http.Request) {
+	})
+	srv := &http.Server{Addr: ":9091", Handler: preStopMux}
+
+	func() { log.Fatal(srv.ListenAndServe()) }()
+}
+
+func reportDowntime(ctx context.Context, clientset *kubernetes.Clientset, downtime int64, failureType string) {
+	klog.Infof("The downtime seen by the pod is %d with type %s", downtime, failureType)
+
+	if os.Getenv("AZURE_CLIENT_ID") != "" {
+		var imageVersion string
+		var err error
+		if *storageType != "" {
+			imageVersion, err = getImageVersion(ctx, clientset, *storageType)
+			if err != nil {
+				klog.Errorf("Error occurred while getting driver version: %v", err)
+			}
+		} else {
+			imageVersion = ""
+		}
+		tester := DowntimeMetric{time.Now().Format(time.RFC3339), *runID, *storageType, *driverVersion, imageVersion, failureType, *workloadType, downtime}
+		bytes, _ := json.Marshal(tester)
```

Comment:
Should we log a fatal error if we cannot marshal to JSON? It would also be good to name this `downtimeMetricJsonBytes` or convert this immediately to a string variable, `downtimeMetricJson`.